src/ucp/core/ucp_context.c

```diff
@@ -1605,6 +1605,11 @@ ucp_add_component_resources(ucp_context_h context, ucp_rsc_index_t cmpt_index,
                         context->reg_md_map[mem_type] |= UCS_BIT(md_index);
                     }
 
+                    if (md_attr->reg_nonblock_mem_types & UCS_BIT(mem_type)) {
```

Comment:
can you please confirm if my understanding is correct:
- `reg_mem_types` is set if blocking, non-blocking registration or both, are supported
  - `reg_md_map` mirrors that
- `reg_nonblock_mem_types` is set if non-blocking is supported
  - `reg_nonblock_md_map` mirrors that
- `reg_block_md_map` is only applicable if mem_type is part of `reg_nb_mem_types`.
  - it is then set if registration is supported but not the non-blocking mode

Reply:
My own understanding is that nonblock registration is a feature of usual (blocking) registration, so `reg_nonblock_mem_types` is always a subset of `reg_mem_types`. And `reg_mem_types` contains mem types for which usual (blocking) registration can be performed by this MD.

Understanding of other fields looks correct FMPOV.